### PR TITLE
Move serialization/storage of privileges to their related roles

### PR DESCRIPTION
**Pull request by seut**
UsersPrivilegesMetadata is deprecated now and privileges of existing ones will be migrated on the fly into the RolesMetadata.

It's helper methods for different privileges operations move to a new PrivilegesModifier helper class.

Extend snapshot restore integration test, to test restoring old UsersPrivilegesMetadata and migrating them inside RolesMetadata.

Relates: #12109

### CODE REVIEW

File: server/src/test/java/io/crate/role/PrivilegesModifierTest.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.crate.role.metadata.RolesMetadata;
+
+public class PrivilegesModifierTest extends ESTestCase {
+
```
Comment (BaurzhanSakhariev) [inline server/src/test/java/io/crate/role/PrivilegesModifierTest.java:44]: Do we need to extends from `ESTestCase`, is it used?

Looks like we are calling only static methods here?

**Review by BaurzhanSakhariev — APPROVED** [review]
Thanks!

LGTM, added some minor comments

**Review by matriv — APPROVED** [review]

**Comment by matriv** [comment]
Thx a lot @seut!
I've enriched the snapshot restore integration test to test restoring old `UsersPrivilegesMetadata` and migrating them into `RolesMetadata`. I've made a small change to `TransportPrivilegesAction` so that `validateRoleNames` also operates on the newly constructed `RolesMetadata`.

**Comment by seut** [comment]
@matriv Great, thx!

**Pull request merged**